### PR TITLE
#861 Stocktake line deletion

### LINKED ISSUE #861: Implement stocktake line deletion functionality
The stocktake line deletion functionality is currently not implemented fully

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #861 

- Adds stocktake line deletion

### CODE REVIEW

File: packages/inventory/src/Stocktake/DetailView/Toolbar.tsx [hunk 1]
```diff
@@ -11,33 +11,19 @@ import {
...
+  useIsStocktakeDisabled,
+  useDeleteSelectedLines,
+} from '../api';
 
 export const Toolbar: FC = () => {
   const t = useTranslation(['distribution', 'common', 'inventory']);
-  const { success, info } = useNotification();
   const isDisabled = useIsStocktakeDisabled();
   const { description, update } = useStocktakeFields('description');
+  const { onDelete } = useDeleteSelectedLines();
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/Toolbar.tsx:24]: shifted everything into this hook

File: packages/inventory/src/Stocktake/api/hooks.ts [hunk 2]
```diff
@@ -159,6 +162,64 @@ export const useSaveStocktakeLines = () => {
   });
 };
 
+export const useDeleteStocktakeLine = () => {
+  const api = useStocktakeApi();
+  return useMutation(api.deleteLines);
+};
+
+export const useDeleteSelectedLines = (): {
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/api/hooks.ts:170]: mainly taken from outbound. 

It seems straight forward enough, though do you think this needs comments? Maybe I'm used to react-query apis and can't see that more commenting could be helpful for others?

Comment (mark-prins) [reply packages/inventory/src/Stocktake/api/hooks.ts:170]: Nice and clear to me - and I really like the hook, and lifting the delete logic out of the Toolbar

**Review by mark-prins — APPROVED** [review]
👍  Looks great!

**Pull request merged**